pep-0725.rst

```diff
@@ -0,0 +1,599 @@
+
+Examples
+--------
+
+These examples show what the ``[external]`` content for a number of packages is
+expected to be.
+
+cryptography 39.0:
+
+.. code:: toml
+
+    [external]
+    build-requires = [
+      "virtual:compiler/rust",
+    ]
+    host-requires = [
+      "pkg:generic/openssl",
+    ]
+
+SciPy 1.10:
+
+.. code:: toml
+
+    [external]
+    build-requires = [
+      "virtual:compiler/c",
+      "virtual:compiler/cpp",
+      "virtual:compiler/fortran",
+      "pkg:generic/ninja",
+    ]
+    host-requires = [
+      "virtual:interface/blas",
+      "virtual:interface/lapack",  # >=3.7.1 (can't express version ranges with PURL yet)
+    ]
+
+    [external.optional-host-requires]
+    dependency_detection = [
+      "pkg:generic/pkg-config",
+      "pkg:generic/cmake",
+    ]
+
+pygraphviz 1.10:
+
+.. code:: toml
+
+    [external]
+    build-requires = [
+      "virtual:compiler/c",
+    ]
+    host-requires = [
+      "pkg:generic/graphviz",
+    ]
+
+NAVis 1.4.0:
+
+.. code:: toml
+
+    [project]
+    optional-dependencies = ["rpy2"]
+
+    [external]
+    build-requires = [
+      "pkg:generic/XCB; platform_system=='Linux'",
+    ]
+
+    [external.optional-dependencies]
+    nat = [
+      "pkg:cran/nat",
+      "pkg:cran/nat.nblast",
+    ]
+
+Spyder 6.0:
+
+.. code:: toml
+
+    [external]
+    dependencies = [
+      "pkg:cargo/ripgrep",
+      "pkg:cargo/tree-sitter-cli",
+      "pkg:golang/github.com/junegunn/fzf",
+    ]
+
+jupyterlab-git 0.41.0:
+
+.. code:: toml
+
+    [external]
+    dependencies = [
+      "pkg:generic/git",
+    ]
+
+    [external.optional-build-requires]
+    dev = [
+      "pkg:generic/nodejs",
+    ]
+
+PyEnchant 3.2.2:
+
+.. code:: toml
+
+    [external]
+    dependencies = [
+      # libenchant is needed on all platforms but only vendored into wheels on
+      # Windows, so on Windows the build backend should remove this external
+      # dependency from wheel metadata.
+      "pkg:github/AbiWord/enchant",
+    ]
+
+
+Backwards Compatibility
+=======================
+
+There is no impact on backwards compatibility, as this PEP only adds new,
+optional metadata. In the absence of such metadata, nothing changes for package
+authors or packaging tooling.
+
+
+Security Implications
+=====================
+
+There are no direct security concerns as this PEP covers how to statically
+define metadata for external dependencies. Any security issues would stem from
+how tools consume the metadata and choose to act upon it.
+
+
+How to Teach This
+=================
+
+External dependencies and if and how those external dependencies are vendored
+are topics that are typically not understood in detail by Python package
+authors. We intend to start from how an external dependency is defined, the
+different ways it can be depended on - from runtime-only with ``ctypes`` or a
+``subprocess`` call to it being a build dependency that's linked against -
```

Comment:
```suggestion
different ways it can be depended on---from runtime-only with ``ctypes`` or a
``subprocess`` call to it being a build dependency that's linked against---
```